### ISSUE TITLE
If there is no storage space available, index.exi file will be deleted

### ISSUE THREAD

**Issue by YukiMatsumura** [google/ExoPlayer#3620]
### Question

`CachedContentIndex.readFile` returns` false` when called with no storage space available.
Does that mean `CachedContentIndex.load` will erase` cached_content_index.exi`?

```
/** Loads the index file. */
public void load() {
    if (!readFile()) {
        atomicFile.delete();
        ...

private boolean readFile() {
    ...
    } catch (IOException e) {
        return false;
```

It is difficult to catch and recover an IoException, but I think it's overkill for download content to be lost just because of no storage space available.

Please tell me if there is a good way to avoid deleting the file. 

### ExoPlayer version
`2.6.0`

thanks.

**Comment by erdemguven** [comment]
In theory, reading shouldn't fail because of no storage space left. Is this something you have seen happened?

**Comment by YukiMatsumura** [comment]
Thank you for a reply.

>  Is this something you have seen happened?

The problem does not have happened. I confirmed the intention of the source code.

> reading should not fail

thanks.
Due to the mechanism of backup/restoration of AtomicFile, I misunderstood that the storage operation is consumed even by the reading operation. 
When I checked it again, it did not seem to be a problem.
I'm sorry to inconvenience you.

My question has been resolved. thanks you 😄
